packages/collector/test/nativeModuleRetry/test.js

```diff
@@ -6,15 +6,48 @@
 'use strict';
 
 const { expect } = require('chai');
-const fs = require('node:fs/promises');
+const fsPromise = require('node:fs/promises');
 const os = require('os');
 const path = require('path');
+const fs = require('fs');
 
 const config = require('../../../core/test/config');
 const { retry } = require('../../../core/test/test_util');
 const ProcessControls = require('../test_util/ProcessControls');
 const globalAgent = require('../globalAgent');
 
+/**
+ * Test suite for verifying the fallback mechanism for loading native add-ons.
+ *
+ * This test ensures that if a native add-on (such as event-loop-stats or gcstats.js) cannot be loaded initially,
+ * the system correctly falls back to using precompiled versions of these add-ons. This ensures the application remains
+ * functions as expected even if the native module is missing or cannot be compiled. The function `attemptRequire`,
+ * defined here: https://github.com/instana/nodejs/blob/main/packages/shared-metrics/src/util/nativeModuleRetry.js#L111
+ * is used for the initial attempt to load the native add-on module. It tries to require the native module from the
+ * root. If this attempt fails, the fallback mechanism is triggered.
+ *
+ * **Issue with npm Workspaces**:
+ *
+ * With `npm workspaces`, dependencies are installed in the root `node_modules` folder instead of individual
+ * package `node_modules` folders. This can complicate managing native modules and their precompiled binaries,
+ * as the fallback mechanism must copy to the correct path. This test verifying that precompiled binaries are copied
+ * and loaded correctly to the @instana/shared-metrics/node_modules folder.
```

Comment:
```suggestion
 * and loaded correctly to the @instana/shared-metrics/node_modules folder. 
 
 The test is **mis-designed**, because the test should not use the local dev environment to check the presence of precompiled binaries. Instead it should create a tmp folder.
```